### PR TITLE
Use optional values for docker memory metrics

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## Proposed commit message
Closes https://github.com/elastic/beats/issues/40060
This deals with an issue where we were reporting docker memory `rss` metrics even though none exist under cgroups V2. This takes the easy way out by just changing the hashmaps to use option types, so we don't report `0` values when none should exist.

I'm not sure there's a better way to do this. As far as I can tell, there's no similar metrics in Docker's cgroups V2 reporting, and in my experience trying to be clever and derive a substitute value usually backfires for heterogeneous monitoring setups. 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Disruptive User Impact

In cases where the docker/memory metrics were reporting an invalid `0`, they will not not report any value at all.

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -203,6 +203,8 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Use namespace for GetListMetrics when exists in AWS {pull}41022[41022]
 - Fix http server helper SSL config. {pull}39405[39405]
 - Fix Kubernetes metadata sometimes not being present after startup {pull}41216[41216]
+- Do not report on-existant 0 values for RSS metrics in docker/memory {pull}41449[41449]
```
Comment (mauri870) [inline CHANGELOG.next.asciidoc:206]:
```suggestion
- Do not report non-existant 0 values for RSS metrics in docker/memory {pull}41449[41449]
```

File: metricbeat/module/docker/memory/memory_test.go [hunk 2]
```diff
@@ -170,7 +195,9 @@ func getMemoryStats(read time.Time, number uint64) types.StatsJSON {
 		},
 	}
 
-	myMemoryStats.MemoryStats.Stats["total_rss"] = number * 5
+	if rssExits {
```
Comment (mauri870) [inline metricbeat/module/docker/memory/memory_test.go:198]: `rssExists`

**Review by rdner — APPROVED** [review]

**Comment by pierrehilbert** [comment]
@fearful-symmetry Do we need this bug fix in 8.17?

**Comment by fearful-symmetry** [comment]
@pierrehilbert It's not really a "fix" as much as a clean-up, so I wouldn't say it's necessary.

**Pull request merged**